internal/services/storage/storage_account_local_user_resource.go

```diff
@@ -0,0 +1,528 @@
+	"github.com/hashicorp/terraform-provider-azurerm/internal/services/storage/parse"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/services/storage/validate"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/pluginsdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/validation"
+	"github.com/hashicorp/terraform-provider-azurerm/utils"
+)
+
+type LocalUserResource struct{}
+
+var _ sdk.ResourceWithUpdate = LocalUserResource{}
+
+type PermissionsModel struct {
+	Create bool `tfschema:"create"`
+	Delete bool `tfschema:"delete"`
+	List   bool `tfschema:"list"`
+	Read   bool `tfschema:"read"`
+	Write  bool `tfschema:"write"`
+}
+type PermissionScopeModel struct {
+	Permissions  []PermissionsModel `tfschema:"permissions"`
+	ResourceName string             `tfschema:"resource_name"`
+	Service      string             `tfschema:"service"`
+}
+type SshAuthorizedKeyModel struct {
+	Description string `tfschema:"description"`
+	Key         string `tfschema:"key"`
+}
+type LocalUserModel struct {
+	HomeDirectory      string                  `tfschema:"home_directory"`
+	Name               string                  `tfschema:"name"`
+	Password           string                  `tfschema:"password"`
+	PermissionScope    []PermissionScopeModel  `tfschema:"permission_scope"`
+	Sid                string                  `tfschema:"sid"`
+	SshAuthorizedKey   []SshAuthorizedKeyModel `tfschema:"ssh_authorized_key"`
+	SshKeyEnabled      bool                    `tfschema:"ssh_key_enabled"`
+	SshPasswordEnabled bool                    `tfschema:"ssh_password_enabled"`
+	StorageAccountName string                  `tfschema:"storage_account_name"`
+}
+
+func (r LocalUserResource) Arguments() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{
+		"name": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ForceNew:     true,
+			ValidateFunc: validate.LocalUserName,
+		},
+		"storage_account_name": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ForceNew:     true,
+			ValidateFunc: validate.StorageAccountName,
+		},
+		"ssh_key_enabled": {
+			Type:         pluginsdk.TypeBool,
+			Optional:     true,
+			Default:      false,
+			AtLeastOneOf: []string{"ssh_key_enabled", "ssh_password_enabled"},
+		},
+		"ssh_password_enabled": {
+			Type:         pluginsdk.TypeBool,
+			Optional:     true,
+			Default:      false,
+			AtLeastOneOf: []string{"ssh_key_enabled", "ssh_password_enabled"},
+		},
+		"home_directory": {
+			Type:     pluginsdk.TypeString,
+			Optional: true,
+		},
+		"ssh_authorized_key": {
+			Type:         pluginsdk.TypeList,
+			Optional:     true,
+			ForceNew:     true,
+			RequiredWith: []string{"ssh_key_enabled"},
+			Elem: &pluginsdk.Resource{
+				Schema: map[string]*pluginsdk.Schema{
+					"description": {
+						Type:     pluginsdk.TypeString,
+						Required: true,
+					},
+					"key": {
+						Type:     pluginsdk.TypeString,
+						Required: true,
```

Comment:
Indeed, I've tried a DSA type SSH key, it can be uploaded successfully when creating the local user, whilst when I `sftp` to it and specified the correct identity file, the client doesn't prompt me to input password for the SSH key, but for password (though I didn't specifiy the password authentication for this local user). Instead, the RSA works as expected. Also I noticed all the example/doc in Azure mentions the RSA, including the autogenerated key is also using RSA. So yes, we'll need to add the validation.